tests/test_signal.py

```diff
@@ -1,24 +1,72 @@
 import random
```

Comment:
Is this file due for a rename to `test_panel.py`?

Reply:
Yes, I can include that here, too.

Reply:
Resolved in cf2f322